src/psyclone/tests/domain/lfric/lfric_extract_driver_creator_test.py

```diff
@@ -35,68 +35,48 @@
 
 ''' This module tests the driver creation for extracted kernels.'''
 
+import os
 import pytest
 
 from psyclone.core import Signature
-from psyclone.domain.lfric import LFRicConstants, LFRicExtractDriverCreator
+from psyclone.domain.lfric import LFRicExtractDriverCreator
 from psyclone.domain.lfric.transformations import LFRicExtractTrans
 from psyclone.errors import GenerationError, InternalError
+from psyclone.parse import ModuleManager
 from psyclone.psyir.nodes import Literal, Routine, Schedule
 from psyclone.psyir.symbols import INTEGER_TYPE
 from psyclone.psyir.tools.dependency_tools import DependencyTools
-from psyclone.tests.utilities import get_invoke
+from psyclone.tests.utilities import Compile, get_base_path, get_invoke
 
 
 API = "dynamo0.3"
 
 
-def test_lfric_driver_constructor():
-    '''Tests the constructor of the LFRic driver creator.'''
-
-    driver_creator = LFRicExtractDriverCreator()
-    default_precision = {"i_def": "int32", "r_def": "real64",
-                         "r_second": "real64", "r_solver": "real32",
-                         "r_tran": "real32"}
-
-    assert driver_creator._precision == default_precision
-
-    # Check that we can modify the default precisions:
-    new_precision = {"r_solver": "real64", "r_tran": "real64"}
-    driver_creator = LFRicExtractDriverCreator(new_precision)
-    assert (driver_creator._precision ==
-            {"i_def": "int32", "r_def": "real64",
-             "r_second": "real64", "r_solver": "real64",
-             "r_tran": "real64"})
-
+@pytest.fixture(scope='function')
+def init_module_manager():
+    ''' The tests in this module all assume that there is no pre-existing
+    ModuleManager object, so this fixture ensures that the module manager
+    instance is deleted before and after each test function. The latter
+    makes sure that any other test executed next will automatically reload
+    the default ModuleManager file.
+    '''
 
-# ----------------------------------------------------------------------------
-def test_lfric_driver_field_mapping():
-    '''Tests that the mapping of fields to precision is as expected.'''
-    mapping = LFRicConstants().DATA_TYPE_MAP
-    correct = {}
-
-    # TODO #2069: use LFRicConstants here
-    for field in ["columnwise_operator",
-                  "field", "integer_field",
-                  "operator", "r_solver_field",
-                  "r_solver_operator", "r_tran_field",
-                  "r_tran_operator"]:
-        correct[mapping[field]["proxy_type"]] = mapping[field]["kind"]
+    infrastructure_path = get_base_path(API)
+    # Define the path to the read-kernel relative to the infrastructure path:
+    read_mod_path = os.path.join(infrastructure_path, "..", "..", "..", "..",
```

Comment:
I didn't get that. os.path.dirname leaves the paths unchanges. I think join is the way to go, or??

Reply:
I meant rather than hardcode the "..":

    # Each os.path.dirname() move up in the folder hierarchy
    BASEPATH = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(
        os.path.dirname(os.path.abspath(__file__))))), "test_files")

Reply:
I **so** do not agree with that ... the replacement:
```
+    psyclone_root = os.path.dirname(os.path.dirname(os.path.dirname(
+        os.path.dirname(os.path.dirname(infrastructure_path)))))
+    read_mod_path = os.path.join(psyclone_root, "lib", "extract", "standalone")
```
is longer, uses 6 function calls instead of 1, is just very hard to understand (don't we all like 5 closing parenthesis next to each other??), and **doesn't even work** if 'infrastructure path' should end with a `/` :
```
os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname("/home/joerg/work/psyclone/src/psyclone/tests/domain/lfric")))))
'/home/joerg/work/psyclone'
os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname("/home/joerg/work/psyclone/src/psyclone/tests/domain/lfric/")))))
'/home/joerg/work/psyclone/src'
```
Admittedly, the path doesn't end with a `/` ... atm :grin:,while my code does work in the latter case:
```
os.path.join("/home/joerg/work/psyclone/src/psyclone/tests/domain/lfric", "..", "..", "..", "..", "..", "lib")
'/home/joerg/work/psyclone/src/psyclone/tests/domain/lfric/../../../../../lib'
os.path.join("/home/joerg/work/psyclone/src/psyclone/tests/domain/lfric/", "..", "..", "..", "..", "..", "lib")
'/home/joerg/work/psyclone/src/psyclone/tests/domain/lfric/../../../../../lib'

```

I even considered stooping so low as to blackmail you by otherwise revealing where you used `"../.."` in a `join` yourself  👿   But seeing that we use this code elsewhere (why oh why) ... I gave in and broke my nice and shiny working code with our standard, risky and fragile approach - we are in for the thrill :)

So, done 😃

Reply:
Wow, that was a big response :-) I agree it doesn't look very nice. I think this is better:

    pdir = infrastructure_path.rstrip("/")
    for idx in range(5):
        pdir = os.path.dirname(pdir)

Possibly we should have our own utility function for this. Anyway, it's not something for this PR.